File: 백준/Bronze/3052. 나머지/나머지.py
a = []
for _ in range(10):
    a.append(int(input()) % 42)
# 10에서 겹치는 수만큼 빼주려고 했다. -> 실패
# cnt = 10
# for i in range(10):
#     for j in range((i+1), len(a)):
#         if a[i] == a[j]:
#             a.pop(a.index(a[j]))
#             cnt -= 1
# print(cnt)
#print(a)

# set을 사용하여 집합의 특성을 사용함.
a = set(a)
print(len(a))

File: 백준/Bronze/3052. 나머지/나머지.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		// 각 index는 수를 의미하며, 값은 나머지로 나온 횟수를 의미한다 1씩 증가 
        // -> 0이 아닌 것들의 개수를 세면 중복을 피해 셀 수 있다.
		int[] remainder = new int[42];
		int cnt = 0;
		Scanner sc = new Scanner(System.in);
		for (int i = 0; i < 10; i++) remainder[sc.nextInt() % 42]++;    // 입력받자마자 나머지를 저장한다.
		for (int i = 0; i < remainder.length; i++) if (remainder[i] != 0) cnt++;
		System.out.println(cnt);
		
	}
}